astropy/units/tests/test_structured.py

```diff
@@ -579,6 +582,39 @@ def test_zeros_ones_like(self, func):
         assert z.shape == self.pv.shape
         assert_array_equal(z, func(self.pv) << self.pv_unit)
 
+    def test_structured_to_unstructured(self):
+        # can't unstructure something with incompatible units
+        with pytest.raises(u.UnitConversionError, match="'km / s'"):
+            rfn.structured_to_unstructured(self.q_pv)
+
+        # it works if all the units are equal
+        struct = u.Quantity((0, 0, 0.6), u.Unit("(eV, eV, eV)"))
+        unstruct = rfn.structured_to_unstructured(struct)
+        assert_array_equal(unstruct, [0, 0, 0.6] * u.eV)
+
+        # also if the units are convertible
+        struct = u.Quantity((0, 0, 0.6), u.Unit("(eV, eV, keV)"))
+        unstruct = rfn.structured_to_unstructured(struct)
+        assert_array_equal(unstruct, [0, 0, 600] * u.eV)
+
+        # and if the dtype is nested
+        struct = [(5, (400.0, 3e6))] * u.Unit('m, (cm, um)')
+        unstruct = rfn.structured_to_unstructured(struct)
+        assert_array_equal(unstruct, [[5, 4, 3]] * u.m)
+
+    def test_unstructured_to_structured(self):
+        # can't structure something that's already structured
+        unstruct = [1, 2, 3] * u.m
+        dtype=np.dtype([("f1", int), ("f2", int), ("f3", int)])
```

Comment:
Absolute and total nitpick in this case, but I like not to encourage `int` for quantity as it generally makes no sense with units; can we just make it `float`?

Reply:
SGTM

Reply:
Done.